examples/src/bin/tlsserver-mio.rs

```diff
@@ -417,79 +416,65 @@ impl OpenConnection {
-    --verbose           Emit log output.
-    --version, -v       Show tool version.
-    --help, -h          Show this screen.
-";
-
-#[derive(Debug, Deserialize)]
+/// Runs a TLS server on :PORT.  The default PORT is 443.
+///
+/// `echo' mode means the server echoes received data on each connection.
+///
+/// `http' mode means the server blindly sends a HTTP response on each connection.
+///
+/// `forward' means the server forwards plaintext to a connection made to `localhost:fport`.
+///
+/// `--certs' names the full certificate chain, `--key' provides the private key.
+#[derive(Debug, Parser)]
 struct Args {
-    cmd_echo: bool,
-    cmd_http: bool,
-    flag_port: Option<u16>,
-    flag_verbose: bool,
-    flag_protover: Vec<String>,
-    flag_suite: Vec<String>,
-    flag_proto: Vec<String>,
-    flag_certs: Option<String>,
-    flag_crl: Vec<String>,
-    flag_key: Option<String>,
-    flag_ocsp: Option<String>,
-    flag_auth: Option<String>,
-    flag_require_auth: bool,
-    flag_no_resumption: bool,
-    flag_tickets: bool,
-    flag_max_early_data: u32,
-    arg_fport: Option<u16>,
+    #[command(subcommand)]
+    cmd: ServerMode,
+    /// Listen on port.
+    #[clap(short, long, default_value = "443")]
+    port: u16,
+    /// Emit log output.
+    #[clap(short, long)]
+    verbose: bool,
+    /// Disable default TLS version list, and use the given versions instead.
+    #[clap(long)]
+    protover: Vec<String>,
+    /// Disable default cipher suite list, and use the given suites instead.
+    #[clap(long)]
+    suite: Vec<String>,
+    /// Negotiate the given protocols using ALPN.
+    #[clap(long)]
+    proto: Vec<String>,
+    /// Read server certificates from the given file. This should contain PEM-format certificates
+    /// in the right order (the first certificate should certify the end entity, matching the
+    /// private key, the last should be a root CA).
+    #[clap(long)]
+    certs: Option<String>,
+    /// Perform client certificate revocation checking using the DER-encoded CRL from the given
+    /// files.
+    #[clap(long)]
+    crl: Vec<String>,
+    /// Read private key from the given file. This should be a private key in PEM format.
+    #[clap(long)]
+    key: Option<String>,
+    /// Read DER-encoded OCSP response from the given file and staple to certificate.
+    #[clap(long)]
+    ocsp: Option<String>,
+    /// Enable client authentication, and accept certificates signed by those roots provided in
+    /// the given file.
+    #[clap(long)]
+    auth: Option<String>,
+    /// Send a fatal alert if the client does not complete client authentication.
+    #[clap(long)]
+    require_auth: bool,
+    /// Disable stateful session resumption.
+    #[clap(long)]
+    no_resumption: bool,
+    /// Support tickets (stateless resumption).
+    #[clap(long)]
+    tickets: bool,
+    /// Support receiving this many bytes with 0-RTT.
+    #[clap(long)]
+    max_early_data: u32,
```

Comment:
```suggestion
    #[clap(long, default_value = "0")]
    max_early_data: u32,
```
(otherwise this becomes mandatory)